proxy/envoy/config.go

```diff
@@ -664,15 +664,22 @@ func buildEgressFromSidecarVirtualHostOnPort(rule *proxyconfig.EgressRule, mesh
 	externalTrafficRoute := buildDefaultRoute(externalTrafficCluster)
 
 	return &VirtualHost{
-		Name:    rule.Domains[0] + "-" + strconv.Itoa(port.Port),
+		Name:    ruleKey + ":" + strconv.Itoa(port.Port),
```

Comment:
why this change ? can you describe the effect here and/or in the PR summary

going from `foo.com-123` to `fooRuleKey:123` (if I understand both the old and new code correctly)

shouldn't it be `host.foo.com:123` - why was it - before ?

Reply:
@ldemailly this one is a commit from https://github.com/istio/pilot/pull/1171, reviewed by @kyessenov. I cherry-picked this commit into this PR for testing, forgot to reset it before submitting a PR. Once https://github.com/istio/pilot/pull/1171 is merged, this commit will not appear in this PR. Sorry for the confusion. 

The idea is to use the key of the rule as a name of the virtual host, and not the first domain of the rule. It is more clear to associate a virtual host with a rule. Since `:` are allowed in the names of Envoy virtual hosts, and this way it is more clear that it is a port, I used `:`.